src/hypothesis/strategies.py

```diff
@@ -945,11 +955,26 @@ def builds(target, *args, **kwargs):
     the target.
 
     """
+    if target_and_args:
+        target, args = target_and_args[0], target_and_args[1:]
+        if not callable(target):
+            raise InvalidArgument('The first positional argument to builds() must be a callable'
+                                  'target to construct.')
+    elif 'target' in kwargs:
+        args = []
+        note_deprecation('Specifying the target as a keyword argument to builds() is deprecated. '
+                         'Provide it as the first positional argument instead.')
+        target = kwargs.pop('target')
+    else:
+        raise InvalidArgument('No target was provided to builds().'
```

Comment:
No space at end of string, and I'd prefer to avoid calling the callable "target".  How about: "builds() must be passed a callable as the first positional argument, but no positional argument was given."

As a general tip, for longer strings we generally start them on the line *after* the opening parenthesis, indented by four spaces.  There are some examples in eg. `decimals()`.

Reply:
>  I'd prefer to avoid calling the callable "target".

Makes sense. Unless you object, I'm still referring to 'target' in the deprecation message ("Specifying the target as a keyword argument to builds() is deprecated ...") since in that case the user just called it `target` so that naming is familiar to them.

Reply:
Yep, I agree that it should be kept in the deprecation message.